fee_allocator/accounting/core_pools.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+from dataclasses import dataclass, field
+from web3 import Web3
+from typing import List, Dict, TYPE_CHECKING
+from decimal import Decimal
+from datetime import datetime
+
+from bal_tools.models import PoolSnapshot, TWAPResult
+from fee_allocator.accounting.interfaces import AbstractCorePool
+from fee_allocator.accounting.overrides import CorePoolOverride, overrides
+
+if TYPE_CHECKING:
+    from fee_allocator.accounting.chains import Chain
+
+
+@dataclass
+class CorePoolData:
+    pool_id: str
+    label: str
+    bpt_price: Decimal
+    tokens_price: List[TWAPResult]
+    gauge_address: str
+    start_snap: PoolSnapshot
```

Comment:
what are these snapshots used for times used for?  start and end of what?

Reply:
start and end of the fee period, defined by the passed in timestamps. snapshots are used for computing the token/bpt over the given period (see CorePoolData methods), same as in v1

Reply:
[d2bf119](https://github.com/BalancerMaxis/protocol_fee_allocator_v2/pull/1/commits/d2bf119bbea828e2fc7cd318f267ac618ede8669)